index.js

```diff
@@ -30,7 +30,7 @@ function throwOnRelease () {
 
 function release (client, err) {
   client.release = throwOnRelease
-  if (err || this.ending) {
+  if (err || this.ending || !client._queryable || client._ending) {
```

Comment:
It would be nice if the queryable state of a client is retrievable via a public method.